### ISSUE TITLE
"Anywhere valid" issue causing data corruption in FP16 to FP8 dtype conversion

### ISSUE THREAD

**Issue by Akash-Nayak** [torch-spyre/torch-spyre#2185]
### Summary
FP8 dtype conversion operations show data corruption when `hidden_dim > 8`: first 8 elements preserved correctly, remaining elements become zeros (50% data loss for `hidden_dim=16`).

### Affected FP8 Operation
- **qfp8ch** (FP16→FP8): 50% data loss when hidden_dim=16

### Reproduction

**Branch Required**: https://github.com/Akash-Nayak/torch-spyre/tree/fp8-quantization-support

**Setup**:
```bash
# Clone and checkout the FP8 support branch
git clone https://github.com/Akash-Nayak/torch-spyre.git
cd torch-spyre
git checkout fp8-quantization-support

# Build torch-spyre
uv pip install -e . --no-deps --no-build-isolation
```

**Test Script** (save as `test_qfp8ch_bug.py`):
```python
#!/usr/bin/env python3
"""Reproduce qfp8ch corruption bug for hidden_dim > 8"""
import torch
import torch_spyre

# Configuration
BATCH_SIZE = 2
SEQ_LEN = 2
HIDDEN_DIM = 16  # Change to 8 to see it work, 16 to see corruption

print(f"Testing qfp8ch with shape: [{BATCH_SIZE}, {SEQ_LEN}, {HIDDEN_DIM}]")

# Create test function
def quantize_fp8(x):
    return torch.ops.spyre.qfp8ch(x)

# Compile with inductor backend
compiled_fn = torch.compile(quantize_fp8, backend="inductor")

# Create test data on Spyre device
x_fp16 = torch.randn(BATCH_SIZE, SEQ_LEN, HIDDEN_DIM, dtype=torch.float16, device='spyre')

# Run compiled function
result = compiled_fn(x_fp16)

# Transfer to CPU for inspection
result_cpu = result.cpu()
x_fp16_cpu = x_fp16.cpu()

# Check for corruption
print(f"\nInput (first sequence, first 16 elements):")
print(x_fp16_cpu[0, 0, :16])

print(f"\nOutput (first sequence, first 16 elements):")
print(result_cpu[0, 0, :16])

# Count zeros in second half
if HIDDEN_DIM > 8:
    second_half = result_cpu[0, 0, 8:16]
    num_zeros = (second_half == 0).sum().item()
    print(f"\nCorruption check: {num_zeros}/{HIDDEN_DIM-8} elements in second half are ZERO")
    
    if num_zeros == HIDDEN_DIM - 8:
        print("❌ BUG REPRODUCED: All elements after index 7 are zeros!")
    elif num_zeros > 0:
        print(f"⚠️  PARTIAL CORRUPTION: {num_zeros} zeros found")
    else:
        print("✅ No corruption detected")
```

**Run**:
```bash
python3 test_qfp8ch_bug.py
```

**Expected Output** (hidden_dim=16):
```
Testing qfp8ch with shape: [2, 2, 16]

Input (first sequence, first 16 elements):
tensor([1.2, 0.5, -0.8, 2.1, 0.3, -1.5, 0.9, 1.7, 0.4, -0.6, 1.1, -0.2, 0.8, -1.3, 0.7, 1.4])

Output (first sequence, first 16 elements):
tensor([1.2, 0.5, -0.8, 2.1, 0.3, -1.5, 0.9, 1.7, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0])
                                                    ↑ Last 8 elements become ZERO

Corruption check: 8/8 elements in second half are ZERO
❌ BUG REPRODUCED: All elements after index 7 are zeros!
```

**Corruption Pattern**:
- hidden_dim=8: ✅ Works (0% corruption)
- hidden_dim=16: ❌ 50% corruption (indices 0-7 OK, 8-15 ZERO)
- hidden_dim=32: ❌ 50% corruption (alternating: [0-7 OK, 8-15 ZERO, 16-23 OK, 24-31 ZERO])

### SDSC Evidence (Failing Case: hidden_dim=16)

**Tensor Definitions**:
```json
"labeledDs_": [
  {
    "ldsIdx_": 0,
    "dsName_": "Tensor0",
    "dsType_": "OUTPUT",           ← Input is OUTPUT type
    "dataFormat_": "SEN169_FP16",
    "stickSize_": [64]             ← 64 FP16 elements per stick
  },
  {
    "ldsIdx_": 1,
    "dsName_": "Tensor1",
    "dsType_": "KERNEL",           ← Output is KERNEL type
    "dataFormat_": "SEN143_FP8",
    "stickSize_": [128]            ← 128 FP8 elements per stick
  }
]
```

**Coordinate Mapping for Output Dimension**
```json
"out": {
  "spatial": 3,
  "temporal": 0,
  "elemArr": 2,                    
  "padding": "nopad",
  "folds": {
    "dim_prop_attr": [
      {"factor_": 1, "label_": "core_fold"},
      {"factor_": 1, "label_": "corelet_fold"},
      {"factor_": 1, "label_": "row_fold"},
      {"factor_": 1, "label_": "elem_arr_1"},     
      {"factor_": 128, "label_": "elem_arr_0"}    
    ]
  }
}
```

**Problem**: The coordinate mapping shows two element arrays (two sticks), but when data is transferred back to CPU, probably only `elem_arr_1` (first stick, indices 0-7) is correctly mapped and the `elem_arr_0` (second stick, indices 8-15) becomes zeros.

### Expected vs Actual Behavior

**Expected**: All FP8 elements preserved correctly regardless of hidden_dim size

**Actual**:
- Single-stick operations (hidden_dim ≤ 8): ✅ Work correctly
- Multi-stick operations (hidden_dim > 8): ❌ Only first stick preserved, rest become zeros

### Impact

- Cannot use qfp8ch for larger hidden_dim
- **Blocks FP8 quantization**

**Comment by manid2** [comment]
This is similar to issue https://github.com/torch-spyre/torch-spyre/issues/2205 except for dtype `fp16 to fp8`.

**Comment by Akash-Nayak** [comment]
The "anywhere valid support" is being tracked in this issue https://github.com/torch-spyre/torch-spyre/issues/2252. So, closing this issue.
